pandas/tests/apply/test_bodo.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+import pytest
+
+from pandas.errors import ExecutionError
+
+import pandas as pd
+import pandas._testing as tm
+
+pytestmark = [pytest.mark.single_cpu, pytest.mark.bodo_udf_engine]
+
+
+@pytest.fixture(params=["bodo"])
+def engine(request):
+    """Test bodo engine by itself to avoid extensions conflicting with numba.
+
+    Note: Using a fixture here to avoid importing at the start of the session.
+    """
+    if request.param == "bodo":
+        pytest.importorskip("bodo")
+    return request.param
```

Comment:
```suggestion
@pytest.fixture
def engine(request):
    """Test bodo engine by itself to avoid extensions conflicting with numba.

    Note: Using a fixture here to avoid importing at the start of the session.
    """
    if request.param == "bodo":
        pytest.importorskip("bodo")
    return "bodo"
```